apps/backend/src/user/user.controller.ts

```diff
@@ -1,8 +1,42 @@
-import { Controller } from '@nestjs/common';
+import { CurrentUser } from '@kir-dev/passport-authsch';
+import { Body, Controller, Get, Param, ParseIntPipe, Patch, Query, UseGuards } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { Role } from '@prisma/client';
 
+import { Roles } from '../auth/decorators/Roles.decorator';
+import { RolesGuard } from '../auth/roles.guard';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { UserEntity } from './entities/user.entity';
 import { UserService } from './user.service';
 
-@Controller('user')
+@Controller('users')
 export class UserController {
   constructor(private readonly userService: UserService) {}
+
+  @Get()
+  @UseGuards(AuthGuard('jwt'), RolesGuard)
+  @Roles(Role.BODY_ADMIN, Role.BODY_MEMBER)
+  async findAll(@Query('page', ParseIntPipe) page: number = 1, @Query('pageSize', ParseIntPipe) pageSize: number = 10) {
+    return this.userService.findMany(page, pageSize);
+  }
+
+  @Get('me')
+  @UseGuards(AuthGuard('jwt'))
+  getCurrentUser(@CurrentUser() user: UserEntity) {
+    return user;
+  }
+
+  @Get(':id')
+  @UseGuards(AuthGuard('jwt'), RolesGuard)
+  @Roles(Role.BODY_ADMIN, Role.BODY_MEMBER)
+  async findOne(@Param('id') id: string) {
+    return this.userService.findOne(id);
+  }
+
+  @Patch(':id')
+  @UseGuards(AuthGuard('jwt'), RolesGuard)
+  @Roles(Role.BODY_ADMIN)
+  async update(@Param('id') id: string, @Body() updateUserDto: UpdateUserDto) {
```

Comment:
I think we need to add the option for the user to update their own profile also. For eg nickname or profile description should be for sure editable. But on the other hand, schresident and vik student should be managed by admins.

Reply:
But these are only for schbody members, right? No point in regular users having nicknames and stuff

Reply:
Ohh, that is fair. I forgot about that

Reply:
so @Dkrisztan I think body members should be able to edit their own nickname and description
and body members and admins should be able to edit the schResident and vikStudent fields (if this is what they'll use during belépőosztás. or maybe I'm confused)

Reply:
The first part makes sense, the second not really . So body members can edit nickname desc and resident and vikstudent fields as well?

Reply:
Since we are not able to check who lives in the dorm and who does not, we agreed on all users will be able to edit their email, sch-status (room number also) as some kind of a self-declaration. Or at least, it is what I remember.
Now the profile page also has a nickname field (for all users), it is very useful for the symetry of the UI :), do you think it is unneccessary?

Reply:
Nope. makes sense I just didn't understand the difference between the admin and member role, regarding this update. This fix will be up later this day.

Reply:
Yes